wgsl/index.bs

```diff
@@ -513,7 +514,205 @@ When unclear from context, this specification indicates
+For example, a future version of WGSL may change the default severity for [=trigger/derivative_uniformity=] from `error` to either `warning` or `info`.
+After such a change to the specification, previously valid programs would remain valid.
+
+### Diagnostic Filtering ### {#diagnostic-filtering}
+
+Once a [=diagnostic=] with a filterable [=diagnostic/triggering rule=] is [=triggered=], WGSL provides mechanisms to discard the diagnostic, or to modify its severity.
+
+A <dfn noexport>diagnostic filter</dfn> *DF* has three parameters:
+* *AR*: a range of source text known as the <dfn noexport>affected range</dfn>
+* *NS*: a new [=diagnostic/severity=]
+* *TR*: a [=diagnostic/triggering rule=]
+
+Applying a diagnostic filter *DF*(*AR*,*NS*,*TR*) to a diagnostic *D* has the following effect:
+* If *D*'s [=diagnostic/triggering location=] is in *AR* and *D*'s [=diagnostic/triggering rule=] is *TR*, then
+        set *D*'s [=diagnostic/severity=] property to *NS*.
+* Otherwise, *D* is unchanged.
+
+A <dfn noexport>range diagnostic filter</dfn> is a [=diagnostic filter=] whose [=affected range=] is a specified
+range of source text.
+A range diagnostic filter is specified as a `@diagnostic` [=attribute=] at the start of the affected source range,
+as specified in the following table.
+A `@diagnostic` attribute [=shader-creation error|must not=] appear anywhere else.
+
+<table class='data'>
+  <caption>Placement of a range diagnostic filter</caption>
+  <thead>
+    <tr><th>Placement<th>Affected Range
+  </thead>
+    <tr><td>Beginning of a [=compound statement=].<td>The compound statement.
+    <tr><td>Beginning of a [=function declaration=].<td>The function declaration.
+    <tr><td>Beginning of an [=statement/if=] statement.<td>The if statement: the [=syntax/if_clause=]
+      and all associated [=syntax/else_if_clause=] and [=syntax/else_clause=] clauses,
+      including all controlling condition expressions.
+    <tr><td>Beginning of a [=statement/switch=] statement.<td>The switch statement: the selector expression and the [=syntax/switch_body=].
+    <tr><td>Beginning of a [=syntax/switch_body=].<td>The [=syntax/switch_body=].
+    <tr><td>Beginning of a [=statement/loop=] statement.<td>The loop statement.
+    <tr><td>Beginning of a [=statement/while=] statement.<td>The while statement: both the condition expression and the loop body.
+    <tr><td>Beginning of a [=statement/for=] statement.<td>The for statement: the [=syntax/for_header=] and the loop body.
+    <tr><td>Immediately before the opening brace (`'{'`) of the [=loop body=] of a [=statement/loop=], [=statement/while=], or [=statement/for=] loop.<td>The [=loop body=].
+    <tr><td>Beginning of a [=syntax/continuing_compound_statement=].<td>The [=syntax/continuing_compound_statement=].
+</table>
+
+Note: A [=function body=], a [=case clause=], a [=default-alone clause=],
+and the bodies of [=syntax/else_if_clause=] and [=syntax/else_clause=] are themselves [=compound statements=].
```

Comment:
Nit: Also the body of an if clause? (this isn't covered by any rule of the table above)